src/ui/mainwindow.cpp

```diff
@@ -455,6 +458,24 @@ void MainWindow::createTab()
     m_tabBar->setCurrentIndex(index);
 }
 
+void MainWindow::duplicateTab()
+{
+    saveTabState();
+    TabState* previous = currentTabState();
+
+    createTab();
```

Comment:
I don't like this approach, creating something just to override. I am thinking about extending `createTab()` with the necessary functionality...